### PR TITLE
[hibernate upgrade] LPS-X hibernate 02 08

### PR DESCRIPTION
**Pull request by yuhai**
@dantewang, I just rebased the pr on master. Compare with https://github.com/dantewang/liferay-portal/pull/2153, I removed LPS-145031 fix.

### CODE REVIEW

**Comment by yuhai** [comment]
ci:test:sf

**Comment by yuhai** [comment]
ci:test:relevant

**Comment by dantewang** [comment]
ci:test

File: portal-impl/src/system.properties [hunk 1]
```diff
@@ -430,4 +430,4 @@
 ## Hibernate
 ##
 
-    hibernate.bytecode.provider=javassist
\ No newline at end of file
+    hibernate.bytecode.provider=bytebuddy
```
Comment (dantewang) [inline portal-impl/src/system.properties:433]: @yuhai If you change it to use bytebuddy here, is it possible to NOT upgrade Javassist in this PR?

Comment (dantewang) [reply portal-impl/src/system.properties:433]: @yuhai For example, if we ask Hibernate to fully use ByteBuddy, is it still needed to upgrade Javassist and fix the class loading issue by adding a lot of import package to bnd files?

Comment (yuhai) [reply portal-impl/src/system.properties:433]: @dantewang,

> If you change it to use bytebuddy here, is it possible to NOT upgrade Javassist in this PR?
yes, I think we may not upgrade Javassist in this PR. I just noticed [5.6.3.Final](https://mvnrepository.com/artifact/org.hibernate/hibernate-core/5.6.3.Final) removed the dependency Javassist .

I will resend the pr to remove Javassist related commits and upgrade hibernate-core to [5.6.5.Final](https://mvnrepository.com/artifact/org.hibernate/hibernate-core/5.6.5.Final

**Pull request closed without merging**